subunit.go

```diff
@@ -166,6 +186,29 @@ func (e *Event) makeTimestamp(c chan<- []byte) {
 	c <- timestamp.Bytes()
 }
 
+func (e *Event) makeMIME(c chan<- []byte) {
+	var mime bytes.Buffer
+	if e.MIME != "" {
+		writeUTF8(&mime, e.MIME)
+	}
+	c <- mime.Bytes()
+}
+
+func (e *Event) makeFileContent(c chan<- []byte) {
+	var fileContent bytes.Buffer
+	if e.FileName != "" {
+		writeUTF8(&fileContent, e.FileName)
+		writeNumber(&fileContent, len(e.FileBytes))
+		fileContent.Write(e.FileBytes)
+	}
+	c <- fileContent.Bytes()
+}
+
+func writeUTF8(b io.Writer, s string) {
```

Comment:
IMO the possible errors here should be taken into account and returned accordingly. 

Also, I think that it would be better if the functions invoked in goroutines have some mean to communicate the possible errors, either with an error channel or returning an instance of a struct with an error field, here are nice examples [1]. 

Other than that looks fine for me. Maybe the subunit.go file is getting big, but all the functionality seems to be tightly related.

[1] https://blog.golang.org/pipelines